snap/component.go

```diff
@@ -47,19 +48,30 @@ type ComponentInfo struct {
 	ComponentSideInfo
 }
 
+// Provenance returns the provenance of the component. This returns
+// naming.DefaultProvenance if no value is set explicitly in the component
+// metadata.
+func (ci *ComponentInfo) Provenance() string {
+	if ci.ComponentProvenance == "" {
+		return naming.DefaultProvenance
+	}
+	return ci.ComponentProvenance
+}
+
 // NewComponentInfo creates a new ComponentInfo.
-func NewComponentInfo(cref naming.ComponentRef, ctype ComponentType, version, summary, description string, csi *ComponentSideInfo) *ComponentInfo {
+func NewComponentInfo(cref naming.ComponentRef, ctype ComponentType, version, summary, description, provenance string, csi *ComponentSideInfo) *ComponentInfo {
```

Comment:
with so many arguments it's getting a bit unclear whether this is more readable than &struct{}

Reply:
I do agree, yeah.